source/agora/node/Runner.d

```diff
@@ -130,12 +130,9 @@ public Listeners runNode (Config config)
     {
         result.registry = new NameRegistry(config.registry, result.node);
         router.registerRestInterface!(NameRegistryAPI)(result.registry);
-        if (config.registry.dns.enabled)
-        {
-            /* auto dnstask = */ runTask(() => runDNSServer(config.registry, result.registry));
-            result.tcp ~= listenTCP(config.registry.dns.port, (conn) => conn.runTCPDNSServer(result.registry),
-                config.registry.dns.address);
-        }
+        /* auto dnstask = */ runTask(() => runDNSServer(config.registry, result.registry));
```

Comment:
Is the `auto dnstask` an implementation for the future ?

Reply:
It's a bit of a tricky issue with the UDP code.
The TCP code spawns fibers for us and our client code run in it. But the UDP code is a bit more ancient and doesn't do that. Instead, we need to manually run the server and listen in it. Which means we need to run a task.
When the node shuts down, we should interrupt that task. However, interrupt throws an Exception, so we would need to handle that. So I've let it for a future enhancement.